lms/djangoapps/courseware/views/views.py

```diff
@@ -720,15 +721,14 @@ def _progress(request, course_key, student_id):
     # additional DB lookup (this kills the Progress page in particular).
     student = User.objects.prefetch_related("groups").get(id=student.id)
 
-    # Fetch course blocks once for performance reasons
-    course_structure = get_course_blocks(student, course.location)
-
-    courseware_summary = grades_progress.summary(student, course, course_structure).chapters
-    if courseware_summary is None:
+    course_grade = CourseGradeFactory.create(student, course)
```

Comment:
I think I would prefer to see more parallelism in the usage of the \*GradeFactories.  If `SubsectionGradeFactory` takes a student in its constructor, and builds grades from an instance of the factory, `CourseGradeFactory` should as well.  Makes it easier for people using the code.